storage/txcache/txListBySenderMap.go

```diff
@@ -0,0 +1,128 @@
+package txcache
+
+import (
+	"sort"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+)
+
+// TxListBySenderMap is
+type TxListBySenderMap struct {
+	Map             *ConcurrentMap
+	Counter         core.AtomicCounter
+	nextOrderNumber core.AtomicCounter
+}
+
+// NewTxListBySenderMap creates a new map-like structure for holding and accessing transactions by sender
+func NewTxListBySenderMap(size int, shardsHint int) TxListBySenderMap {
+	// We'll hold at most "size" lists of at least 1 transaction
+	backingMap := NewConcurrentMap(size, shardsHint)
```

Comment:
same as above.

Reply:
Actually, `TxListBySenderMap` wants to have full control over the map (and its associated counter). It wants to construct it by itself. But I have made it private and renamed it to `backingMap`. Is it ok?